zpi-frontend/src/pages/Admin/Thesis/Details.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import { Thesis } from '../../../models/Thesis';
+import { Program } from '../../../models/Program';
+import api from '../../../utils/api';
+import useAuth from "../../../auth/useAuth";
+import handleSignOut from "../../../auth/Logout";
+import { useTranslation } from "react-i18next";
+import DeleteConfirmation from '../../../components/DeleteConfirmation';
+import { toast } from 'react-toastify';
+
+const ThesisDetails: React.FC = () => {
+  // @ts-ignore
+  const { auth, setAuth } = useAuth();
+  const { i18n, t } = useTranslation();
+  const navigate = useNavigate();
+
+  const { id } = useParams<{ id: string }>();
+  const [thesis, setThesis] = useState<Thesis>();
+
+  useEffect(() => {
+    const response = api.get(`http://localhost:8080/thesis/${id}`)
+      .then((response) => {
+        setThesis(response.data);
+      })
+      .catch((error) => {
+        console.error(error);
+        if (error.response.status === 401 || error.response.status === 403) {
+          setAuth({ ...auth, reasonOfLogout: 'token_expired' });
+          handleSignOut(navigate);
+        }
+      });
+  }, [id]);
+
+  const [programs, setPrograms] = useState<Program[]>([]);
+  useEffect(() => {
+    api.get('http://localhost:8080/program')
+      .then((response) => {
+        setPrograms(response.data);
+        console.log(programs);
+      })
+      .catch((error) => {
+        console.error(error);
+        if (error.response.status === 401 || error.response.status === 403) {
+          setAuth({ ...auth, reasonOfLogout: 'token_expired' });
+          handleSignOut(navigate);
+        }
+      });
+  }, []);
+
+  const [expandedPrograms, setExpandedPrograms] = useState<number[]>([]);
+
+  const toggleProgramExpansion = (programId: number) => {
+    if (expandedPrograms.includes(programId)) {
+      setExpandedPrograms(expandedPrograms.filter(id => id !== programId));
+    } else {
+      setExpandedPrograms([...expandedPrograms, programId]);
+    }
+  };
+
+  const [showDeleteConfirmation, setShowDeleteConfirmation] = useState(false);
+
+  const handleDeleteClick = () => {
+    setShowDeleteConfirmation(true);
+  };
+
+  const handleConfirmDelete = () => {
+    api.delete(`http://localhost:8080/thesis/${id}`)
```

Comment:
Nie ma takiej metody na backendzie, więc nie da się tego w pełni przetestować, ale wygląda jakby było ok

Reply:
Wiem, no ale powinna być i użyłam jej analogicznie do reszty deletów, więc jak na backendzie się pojawi to powinno dzialać.

Reply:
Oki, w takim razie spoko